main.go

```diff
@@ -306,19 +306,29 @@ func mainerr() error {
 					md = filepath.Join(md, "@v", emv)
 				}
 
-				epp, err := module.EncodePath(mp.ImportPath)
-				if err != nil {
-					return fmt.Errorf("failed to encode package relative path %v: %v", mp.ImportPath, err)
+				if mp.Module.Path != mp.ImportPath {
+					// We don't need to encode what remains in the pkg path because
+					// we've already uniquely identified the module. If there are
+					// case flips within a module then...  well, we'll see.
+					pkgRem := strings.TrimPrefix(mp.ImportPath, mp.Module.Path+"/")
+					mainrel = filepath.Join(md, filepath.FromSlash(pkgRem))
```

Comment:
shouldn't this be `path`, not `filepath`? since it's a module import path, and not a path of a directory on disk.

Reply:
`filepath` is correct here I think. `md` is the file path of the encoded module path. Hence it only makes sense to join another filepath.